doc/development/submitting/general.md

```diff
@@ -0,0 +1,262 @@
+they are managed by SVN itself.
+
+If you are not sure if your style is correct, ask on mailing list.
+
+Some **bad examples** (of single line commit messages):
+
+```txt
+r.slope.aspect: fix compilation
+(missing information what exactly was broken and reasoning behind the fix; it's
+clear that we are not trying to break something by the commit)
+
+wxGUI/render: attempt to fix #560
+(missing information what is #560 and how we are trying to fix it)
+
+Add tests for Table and Columns classes
+(we don't know where the classes belongs to, prefix pygrass: or
+pythonlib/pygrass would tell us in this case)
+
+fix bug introduced in r60216
+(missing information how the bug was fixed and which bug it was)
+
+fix r60216 (i18n)
+(it should probably say something like: wxGUI: fix insufficient handling of i18n
+(introduced in r60216))
+
+libraster:Added raster name and row info to get/put\_row error messages
+(missing space after colon)
+
+d.histogram launched from map display toolbar doesn't work
+(this is description of what is wrong, not how it is fixed)
+
+fixed loading from a file, should I backport it?
+(commit messages are not for opening discussions or general communication)
+```
+
+### Creating (legacy) diffs
+
+Be sure to create unified (`diff -u`) format. "Plain" diffs (the default format)
+are risky, because they will apply without warning to code which has been
+substantially changed; they are also harder to read than unified.
+
+Such diffs should be made from the top-level directory, e.g.
+`git diff display/d.vect/main.c`; that way, the diff will include the pathname
+rather than just an ambiguous `main.c`.
+
+### Comments
+
+PLEASE take the time to add comments throughout your code explaining what the
+code is doing. It will save a HUGE amount of time and frustration for other
+programmers that may have to change your code in the future.
+
+### End Of Line
+
+Make sure a new line is at the end of each file and UNIX style newlines are
+used (`\n`).
+
+### Branches and backports
+
+The GRASS GIS repository on GitHub has the "main" and several release branches.
+All the development should happen in "main" (the development branch). All the other
+branches are usually associated with releases and should contain stable code
+which is being prepared towards the given release.
+
+When a bug is fixed the fix should be committed to "main", tested there, and then
+backported to the release branch or branches, if appropriate. The testing before
+doing a backport should include user testing, running of automated test (if
+available), and compilation of the whole source tree (ideally after the
+`make distclean` step). Note that testing should already be done also prior to
+the original commit to "main". Also note that not all these steps have to be
+done manually, you can take an advantage of our [​CI](https://github.com/OSGeo/grass/actions)
+or [​automated runs of the GRASS GIS testing suite](http://fatra.cnr.ncsu.edu/grassgistests/).
```

Comment:
```suggestion
done manually, you can take an advantage of our [​CI](https://github.com/OSGeo/grass/actions).
```

I suppose this solution was what you reached for?!

Reply:
Yes!